assembly.py

```diff
@@ -471,6 +473,53 @@ def order_and_orient(inFasta, inReference, outFasta,
     #    os.unlink(trimmed)
     return 0
 
+def _call_order_and_orient_orig(inReference, outFasta, outAlternateContigs, **kwargs):
+    _order_and_orient_orig(inReference=inReference, outFasta=outFasta, outAlternateContigs=outAlternateContigs, **kwargs)
+
+def order_and_orient(inFasta, inReference, outFasta,
```

Comment:
Hm.. one thing. I kind of wonder if it might be preferable to have inReference be a list of fasta files (with argparse nargs='+')? Still thinking about this....

Reply:
Can allow this and just concat all the listed fastas into one, while also permitting multiple references in each fasta.

Reply:
True... but then maybe you'd want to set `n_genome_segments` automatically based on the pre-concatenated inputs? If you think it's straightforward to make the CLI flexible either way, that's cool...

Reply:
Do we have any actual CLI users we can get input from, as to what CLI would be less confusing?  For pipeline users n_genome_segments gets set automatically from the config file.

Reply:
While the CLI may not have a lot of direct users for day to day use (more for one-off small analyses), the CLI is the common denominator between multiple ways of executing pipelines on viral-ngs. The snakemake rules are the oldest in our codebase, but even since their inception, I have never considered them sacred or primary. DNAnexus and Cromwell are both happy with the idea of giving users typed collections for inputs (e.g. `Array[File]`) and since argparse is too, why not let them play together?

I'm thinking more generally of the web user who wants to try a few references, whether it'd be easier to have them just specify a list of fasta files (each referring to one ref genome) or to have them concatenate them all to one file ahead of time (in the same order, etc).

For the lastal step, we have them concatenate, but there are two reasons why. One is that the lastal db often has a huge number of genomes (so it'd be crazy to have them specify hundreds of input files), and the other is that the ordering within the file doesn't matter (so there's nothing for them to mess up in terms of how they put it together).

In the case of multi-reference selection, I could imagine that most use cases (initially at least) would involve selecting from a small handful (maybe less than a dozen) options that span known phylogentic diversity within a species of interest. Which is why I was thinking that exposing this as a list of fasta files might make sense. And in order to expose it at the pipeline level, it's easier if the CLI supports it. We'll have to see how the usage evolves over time, but I initially lean towards having the user specify a list of fastas, at least for the web UI.

Reply:
On the command line you can already use <(cat ref1.fasta ref2.fasta ...) for inReference

Reply:
There's already a consistency check in Python that the number of sequences in inReference is a multiple of n_genome_segments, if n_genome_segments is specified.  If n_genome_segments is not specified, we assume (for backwards compatibility) that inReference contains one reference, so its segment count gives n_genome_segments.   If n_genome_segments is not specified and multiple inReference files are listed, we can assume that each contains one genome and that'll give the n_genome_segments.

Maybe the CLI could recognize a config file from which it could get some parameters if they're not specified on the command-line.  Right now the config file is tied to Snakemake, but loading some parameters makes sense outside that context, e.g. threads, tempdir, n_genome_segments, or other parameters that make sense across several commands.  Or default parameters for a specific CLI command could be specified in a config file.  There could be a project config file, which overrides the user config file, which overrides the system config file, as for .condarc etc.

Reply:
- if we cat multiple fastas together, there's no guarantee that each individual fasta has the right number of segments (only that the sum of them is a multiple of n_genome_segments), unless we enforce it outside of python (e.g. in the shell prior to invocation of viral-ngs)
- I don't like the idea of a CLI-level config file, I have trouble envisioning where that would make sense, and would much rather expose them as argparse params where sensible. The proper default value for `threads` is not at all universal across all the commands and use cases, the proper default value for `--tmp_dir` is already some subdirectory of `$TMPDIR` (in keeping with the behavior of most unix tools). The proper conda defaults should be handled at a layer of abstraction above the CLI. `n_genome_segments` is not only project specific, but potentially sample-specific within a project.